packages/docs/site/docs/11-architecture/18-host-your-own-playground.md

```diff
@@ -0,0 +1,169 @@
+
+Also, if you expect to see heavy traffic, it's good etiquette to provide your own server/CDN resources to serve the static file assets, rather than relying on the `wordpress.net` server.
+
+#### Usage
+
+A self-hosted Playground can be embedded as an iframe.
+
+```html
+<iframe src="https://my-playground.com"></iframe>
+```
+
+Or dynamically loaded by passing the remote URL to the Playground Client.
+
+```ts
+import { startPlaygroundWeb } from '@wp-playground/client';
+
+const client = await startPlaygroundWeb({
+	iframe: document.getElementById('wp'),
+	remoteUrl: `https://my-playground.com/remote.html`,
+});
+```
+
+## Build static assets
+
+Create a shallow clone of the Playground repository.
+
+```sh
+git clone -b trunk --single-branch --depth 1 git@github.com:WordPress/wordpress-playground.git
+```
+
+Install dependencies, and build the website.
+
+```sh
+npm install
+npm run build:website
+```
+
+This command internally runs the `nx` task `build:wasm-wordpress-net`. It copies the built assets from packages `remote` and `website` into a new folder at the path.
+
+```
+dist/packages/playground/wasm-wordpress-net
+```
+
+The entire service of the Playground consists of the content of this folder.
+
+It includes:
+
+-   Data and WASM files for all available PHP and WordPress versions
+-   `remote.html` - the core of Playground
+-   `index.html` - the shell, or browser chrome
+-   Web Worker script
+
+You can deploy the content of the folder to your server using SSH, such as `scp` or `rsync`.
+
+It is a static site, except for these dynamic aspects.
+
+-   Apache server directive `.htaccess` file - the result of combining `.htaccess` from the packages `remote` and `website`
+-   Plugin download proxy, `plugin-proxy.php`
+
+For these to work, you need a server environment with Apache and PHP installed.
+
+## NGINX configuration
+
+As an alternative to Apache, here is an example of using NGINX to serve the Playground.
+
+The Apache `.htaccess` file looks like this.
+
+```htaccess
+AddType application/wasm .wasm
+AddType	application/octet-stream .data
+
+<FilesMatch "iframe-worker.html$">
+  Header set Origin-Agent-Cluster: ?1
+</FilesMatch>
+
+Header set Cross-Origin-Resource-Policy: cross-origin
+Header set Cross-Origin-Embedder-Policy: credentialless
+RewriteEngine on
+RewriteRule ^plugin-proxy$ plugin-proxy.php [NC]
+```
+
+An equivalent in NGINX.
+
+```nginx
+location ~* .wasm$ {
+  types {
+    application/wasm wasm;
+  }
+}
+
+location ~* .data$ {
+  types {
+    application/octet-stream data;
+  }
+}
+
+location /iframe-worker.html {
+  add_header Origin-Agent-Cluster ?1;
+}
+
+location /plugin-proxy {
+  try_files plugin-proxy.php /plugin-proxy.php$is_args$args;
+  include fastcgi_params;
+  fastcgi_pass php;
+}
+
+location /scope:.* {
+  rewrite ^scope:.*?/(.*)$ $1 last;
+}
```

Comment:
Note this rule isn't there in .htaccess

Reply:
OK, removed.

Reply:
Just now I ran into a situation where this server directive for scoped URL is useful. It reminded me why I included this in the NGINX config. There used to be an equivalent in the `.htaccess` file.

In the embedded website's Gutenberg editor, when clicking Preview -> Preview in new tab, it opens a new browser tab with a scoped URL, like `https://example.com/scope:0.xxx/...`. For some reason it's working by default with `playground.wordpress.net`, but for my self-hosted playground, I needed this NGINX `location` clause to rewrite them to unscoped URLs.

Reply:
For this one, I restored the NGINX rule for handling scoped URLs because it was necessary in my case, for example, when Gutenberg opens a Preview in a new browser tab with a scoped URL.

I'm hesitant to resolve this conversation, as I don't understand why it's necessary for NGINX but not for Apache. Maybe I can create an issue as a reminder to look into it deeper.